pennylane/math/quantum.py

```diff
@@ -972,6 +975,101 @@ def sqrt_matrix(density_matrix):
     return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ qml.math.conj(qml.math.transpose(vecs))
 
 
+def sqrt_matrix_sparse(density_matrix):
+    r"""Compute the square root matrix of a sparse density matrix where :math:`\rho = \sqrt{\rho} \times \sqrt{\rho}`
+
+    Args:
+        density_matrix (sparse): 2D sparse density matrix of the quantum system.
+
+    Returns:
+        (sparse): Square root of the density matrix. Even though data type as `csr_matrix` or `csc_matrix`, the output matrix is not guaranteed to be sparse as well.

```

Comment:
I'm confused between the given type of the output and the statement of this sentence :sweat_smile: Is it a sparse output or not?
Also, is there a slight grammar hickup in the second part? Like, should it be
 ```suggestion
        (sparse): Square root of the density matrix. Even for data types like `csr_matrix` or `csc_matrix`, the output matrix is not guaranteed to be sparse as well.

```

Reply:
Oh sorry for my English. Here what I wanted to say is that the input is sparse intrinsicly , the output is `scipy.sparse` as well, but actually they may have dense nature, i.e. nnz close to their n^2

Reply:
Ahhh, sorry, I was set on the "dtype meaning" of the word sparse and forgot about the actual meaning of the word here :see_no_evil: 
Thanks for clarifying!